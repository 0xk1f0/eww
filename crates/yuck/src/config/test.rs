use crate::{
    config::config::Config,
    parser::{self, ast::Ast, from_ast::FromAst, lexer::Lexer},
};

#[test]
fn test_config() {
    let input = r#"
        (defwidget foo [arg]
            "heyho")
        (defwidget bar [arg arg2]
            "bla")
        (defvar some_var "bla")
        (defpollvar stuff :interval "12s" "date")
        (deftailvar stuff "tail -f stuff")
        (defwindow some-window
                   :stacking "fg"
                   :monitor 12
                   :resizable true
                   :geometry (geometry :width "12%" :height "20px")
                   :reserve (struts :side "left" :distance "30px")
            (foo :arg "bla"))
    "#;

    let lexer = Lexer::new(0, input.to_string());
    let p = parser::parser::ToplevelParser::new();
    let (span, parse_result) = p.parse(0, lexer).unwrap();
    // TODO implement another YuckFiles thing to test here again
    // let config = Config::from_ast(Ast::List(span, parse_result));
    // insta::with_settings!({sort_maps => true}, {
    // insta::assert_ron_snapshot!(config.unwrap());
    //});
}
