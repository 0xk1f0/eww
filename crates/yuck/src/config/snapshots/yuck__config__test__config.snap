---
source: crates/yuck/src/config/test.rs
expression: config.unwrap()

---
Config(
  widget_definitions: {
    "bar": WidgetDefinition(
      name: "bar",
      expected_args: [
        AttrName("arg"),
        AttrName("arg2"),
      ],
      widget: WidgetUse(
        name: "foo",
        attrs: Attributes(
          span: Span(51, 61, 0),
          attrs: {
            AttrName("arg"): AttrEntry(
              key_span: Span(52, 56, 0),
              value: SimplExpr(Span(57, 61, 0), Literal(DynVal("hi", Span(57, 61, 0)))),
            ),
          },
        ),
        children: [],
        span: Span(47, 62, 0),
        name_span: Span(48, 51, 0),
      ),
      span: Span(9, 63, 0),
      args_span: Span(24, 34, 0),
    ),
  },
  window_definitions: {
    "some-window": WindowDefinition(
      name: "some-window",
      geometry: Some(WindowGeometry(
        anchor_point: AnchorPoint(
          x: START,
          y: START,
        ),
        offset: Coords(
          x: Pixels(0),
          y: Pixels(0),
        ),
        size: Coords(
          x: Percent(12),
          y: Pixels(20),
        ),
      )),
      stacking: Foreground,
      monitor_number: Some(12),
      widget: WidgetUse(
        name: "bar",
        attrs: Attributes(
          span: Span(467, 478, 0),
          attrs: {
            AttrName("arg"): AttrEntry(
              key_span: Span(468, 472, 0),
              value: SimplExpr(Span(473, 478, 0), Literal(DynVal("bla", Span(473, 478, 0)))),
            ),
          },
        ),
        children: [],
        span: Span(463, 479, 0),
        name_span: Span(464, 467, 0),
      ),
      resizable: true,
      backend_options: BackendWindowOptions(
        wm_ignore: false,
        sticky: true,
        window_type: Dock,
        struts: StrutDefinition(
          side: Left,
          dist: Pixels(30),
        ),
      ),
    ),
  },
  var_definitions: {
    VarName("some_var"): VarDefinition(
      name: VarName("some_var"),
      initial_value: DynVal("bla", Span(89, 94, 0)),
      span: Span(72, 95, 0),
    ),
  },
  script_vars: {
    VarName("stuff"): Listen(ListenScriptVar(
      name: VarName("stuff"),
      command: "tail -f stuff",
      command_span: Span(168, 183, 0),
      name_span: Span(162, 167, 0),
    )),
  },
)
